### ISSUE TITLE
False-positive for react/display-name after 7.29.4 → 7.30.0 upgrade

### ISSUE THREAD

**Issue by angelikatyborska** [jsx-eslint/eslint-plugin-react#3300]
After upgrading eslint-plugin-react from 7.29.4 to 7.30.0 a new linting error appeared in my project that doesn't seem right. My project is a big full-stack monolith web app and it contains both Vue.js and React.js code. Now I am getting `react/display-name` for some Vue/Vuex code that has nothing to do with React. I stripped down the file, the original was much longer, but this still triggers the error:

```js
import Vuex from 'vuex'
import Vue from 'vue'

// code omitted

const getters = {
  getChat: state => (scopePath, partnerId) => {
    const selectedChatScope = state.chats && state.chats.find(chat => chat.path === scopePath)

    if (!selectedChatScope) {
      return null
    }

    return 'something'
  }
}

export {
  getters
}
```

```
  7:21  error  Component definition is missing display name  react/display-name
```

**Comment by ljharb** [comment]
cc @golopot
